oxide/crates/core/src/lib.rs

```diff
@@ -30,6 +32,173 @@ pub struct ChangedContent {
     pub extension: String,
 }
 
+#[derive(Debug, Clone)]
+pub struct ContentPathInfo {
+    pub base: String,
+}
+
+// Goals:
+//
+// 1. Recursively watch all top-level folders but not the root folders.
+// 2. Watch files in root directly.
+// 3. Watch files in the `public` folder directly (and don't watch the `public` folder itself).
+// 4. Always watch a common set of known file extensions
+//    (js,jsx,ts,tsx,html,php,vue,svelte,astro,md,mdx,...).
+// 5. Merge known list of file extensions with discovered file extensions in the project (marko).
+
+pub fn resolve_content_paths(args: ContentPathInfo) -> Vec<String> {
+    let root = Path::new(&args.base);
+
+    // A list of direct folder names where we can't use globs, but we should track each file individually instead.
+    // This is because these folders are often used for "src" and also "dist" like files.
+    let static_direct_folder_indentifiers = vec!["public"];
+
+    // A list of known extensions + a list of extensions we found in the project.
+    let mut found_extensions = FxHashSet::from_iter(
+        include_str!("fixtures/template-extensions.txt")
+            .trim()
+            .lines()
+            .filter(|x| !x.starts_with('#'))
+            .filter(|x| *x != ""),
```

Comment:
It seems these 2 filters can be merged, right?

Reply:
Totally can! But since there is basically no performance hit, I prefer the readability instead because now I don't have to think about the correct `&&` vs `||` to use and see if I can apply De Morgan's law correctly if needed 😅